In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# csv파일 공공데이터 (사용하지 않을 예정)

In [ ]:
import pandas as pd

In [ ]:
data = pd.read_csv('data/방위사업청_국방통합 용어사전_20211231 (1).csv', encoding = 'cp949')
data.head()

순번               표제어                         분류  \
0   1  "37" 국가 재고 번호 품목               사업/계약관리/국제계약   
1   2             1.4후퇴  전쟁/군사정보>군사 용어사전>전쟁,군사용어사전   
2   3       127밀리 함포 탄약                  무기체계/기동화력   
3   4          1차 협력 업체               사업/계약관리/계약관리   
4   5          2S19 자주포                  무기체계/기동화력   

                              영문  \
0     National Stock Number Item   
1                            NaN   
2       127milli Naval Gun Shell   
3           1st Cooperative Firm   
4  2S19 Self-propelled Artillery   

                                                  설명                    출처  \
0  ‘37’국가재고번호는 대한민국 군수품의 국가재고번호로서 나토(NATO)에서 부여한 ...  방사청용어국방부군수전력용어사전KIMA   
1  중공군의 개입으로 인하여 1951년 1월 4일 한국전쟁 중 두 번째로 서울을 포기한...             국방부 전쟁기념관   
2  08년도부터 전력화된 해군의 주력함포인 127밀리 함포용 탄약으로 '05~'08기간...                 방사청용어   
3          계약과 관련하여 계약상대자로부터 하도급을 받은 자연인 또는 법인을 말한다.         방위사업청 예규제148호   
4  러시아에서 개발한 최대사거리 29km, 전투중량 42톤, 방열 및 사격통제를 수동 ...                 방사청용어   

                                      출처1  
0  국방전력발전업무훈령, 개정 2012.  2. 3.  훈령 제1388호  
1                          연합.합동작전 군사용어사전  
2                                 방사청홈페이지  
3                   일반무기체계 연구개발 계약특수조건 표준  
4                                 방사청홈페이지

In [ ]:
data_category_filter = data[data['분류'].str.split('/').str[0] == '무기체계']
data_category_filter = data_category_filter.drop(['순번','출처1'],axis=1)
data_category_filter

표제어         분류                                      영문  \
2            127밀리 함포 탄약  무기체계/기동화력                127milli Naval Gun Shell   
4               2S19 자주포  무기체계/기동화력           2S19 Self-propelled Artillery   
8            30mm 자주 대공포  무기체계/기동화력  Self-Propelled AA 30mm Twin-Gun System   
11     40mm/L70(T)2연장 함포    무기체계/함정               40㎜ Twin Naval Gun System   
14                  88전차  무기체계/기동화력                                 88-Tank   
...                  ...        ...                                     ...   
18304                홍상어    무기체계/함정                                     NaN   
18600                후속함    무기체계/함정                                     NaN   
18635       휴대용 지대공 유도무기  무기체계/유도무기                                  KP-SAM   
18644                 흑룡    무기체계/함정                                     NaN   
18650             흑표(전차)  무기체계/기동화력                                     NaN   

                                                      설명  \
2      08년도부터 전력화된 해군의 주력함포인 127밀리 함포용 탄약으로 '05~'08기간...   
4      러시아에서 개발한 최대사거리 29km, 전투중량 42톤, 방열 및 사격통제를 수동 ...   
8      30mm자주대공포는 K-30 비호의 사업명으로 국내 기술로 개발된 30mm 쌍열 대...   
11                                                  ☞ 노봉   
14                                 K1전차의 통상명칭이다. ☞ K1 전차   
...                                                  ...   
18304  장거리 대잠어뢰(홍상어)는 KD 급 이상의 구축함에 장착하여 원거리 적 잠수함을 타...   
18600  선도함 이후 후속적으로 건조되는 동형 함정을 말하며, 후속함은 초도물량(Low Ra...   
18635  한국이 국내 기술로 설계·제작한 한국형 휴대용 대공유도무기(KP-SAM)로, 국산화...   
18644  수상함에 탑재하는 예인음탐기이다. 흑룡은 적 잠수함을 조기에 탐지, 식별, 추적할 ...   
18650                              K2전차의 통상명칭이다. ☞ K2 전차   

                                    출처  
2                                방사청용어  
4                                방사청용어  
8                                방사청용어  
11                               방사청용어  
14                               방사청용어  
...                                ...  
18304                            방사청용어  
18600  방사청용어+과학기술용어사전+국방부군수전력용어사전+KIMA  
18635                            방사청용어  
18644                            방사청용어  
18650                            방사청용어  

[358 rows x 5 columns]

In [ ]:
data_category_filter.reset_index(drop = True)

표제어         분류                                      영문  \
0          127밀리 함포 탄약  무기체계/기동화력                127milli Naval Gun Shell   
1             2S19 자주포  무기체계/기동화력           2S19 Self-propelled Artillery   
2          30mm 자주 대공포  무기체계/기동화력  Self-Propelled AA 30mm Twin-Gun System   
3    40mm/L70(T)2연장 함포    무기체계/함정               40㎜ Twin Naval Gun System   
4                 88전차  무기체계/기동화력                                 88-Tank   
..                 ...        ...                                     ...   
353                홍상어    무기체계/함정                                     NaN   
354                후속함    무기체계/함정                                     NaN   
355       휴대용 지대공 유도무기  무기체계/유도무기                                  KP-SAM   
356                 흑룡    무기체계/함정                                     NaN   
357             흑표(전차)  무기체계/기동화력                                     NaN   

                                                    설명  \
0    08년도부터 전력화된 해군의 주력함포인 127밀리 함포용 탄약으로 '05~'08기간...   
1    러시아에서 개발한 최대사거리 29km, 전투중량 42톤, 방열 및 사격통제를 수동 ...   
2    30mm자주대공포는 K-30 비호의 사업명으로 국내 기술로 개발된 30mm 쌍열 대...   
3                                                 ☞ 노봉   
4                                K1전차의 통상명칭이다. ☞ K1 전차   
..                                                 ...   
353  장거리 대잠어뢰(홍상어)는 KD 급 이상의 구축함에 장착하여 원거리 적 잠수함을 타...   
354  선도함 이후 후속적으로 건조되는 동형 함정을 말하며, 후속함은 초도물량(Low Ra...   
355  한국이 국내 기술로 설계·제작한 한국형 휴대용 대공유도무기(KP-SAM)로, 국산화...   
356  수상함에 탑재하는 예인음탐기이다. 흑룡은 적 잠수함을 조기에 탐지, 식별, 추적할 ...   
357                              K2전차의 통상명칭이다. ☞ K2 전차   

                                  출처  
0                              방사청용어  
1                              방사청용어  
2                              방사청용어  
3                              방사청용어  
4                              방사청용어  
..                               ...  
353                            방사청용어  
354  방사청용어+과학기술용어사전+국방부군수전력용어사전+KIMA  
355                            방사청용어  
356                            방사청용어  
357                            방사청용어  

[358 rows x 5 columns]

In [ ]:
data_category_filter = data_category_filter.reset_index(drop = True)

In [ ]:
data_category_filter.to_csv('data/공공데이터_무기체계_필터링.csv', index = False)

In [ ]:
pd.read_csv('data/공공데이터_무기체계_필터링.csv')

표제어         분류                                      영문  \
0          127밀리 함포 탄약  무기체계/기동화력                127milli Naval Gun Shell   
1             2S19 자주포  무기체계/기동화력           2S19 Self-propelled Artillery   
2          30mm 자주 대공포  무기체계/기동화력  Self-Propelled AA 30mm Twin-Gun System   
3    40mm/L70(T)2연장 함포    무기체계/함정               40㎜ Twin Naval Gun System   
4                 88전차  무기체계/기동화력                                 88-Tank   
..                 ...        ...                                     ...   
353                홍상어    무기체계/함정                                     NaN   
354                후속함    무기체계/함정                                     NaN   
355       휴대용 지대공 유도무기  무기체계/유도무기                                  KP-SAM   
356                 흑룡    무기체계/함정                                     NaN   
357             흑표(전차)  무기체계/기동화력                                     NaN   

                                                    설명  \
0    08년도부터 전력화된 해군의 주력함포인 127밀리 함포용 탄약으로 '05~'08기간...   
1    러시아에서 개발한 최대사거리 29km, 전투중량 42톤, 방열 및 사격통제를 수동 ...   
2    30mm자주대공포는 K-30 비호의 사업명으로 국내 기술로 개발된 30mm 쌍열 대...   
3                                                 ☞ 노봉   
4                                K1전차의 통상명칭이다. ☞ K1 전차   
..                                                 ...   
353  장거리 대잠어뢰(홍상어)는 KD 급 이상의 구축함에 장착하여 원거리 적 잠수함을 타...   
354  선도함 이후 후속적으로 건조되는 동형 함정을 말하며, 후속함은 초도물량(Low Ra...   
355  한국이 국내 기술로 설계·제작한 한국형 휴대용 대공유도무기(KP-SAM)로, 국산화...   
356  수상함에 탑재하는 예인음탐기이다. 흑룡은 적 잠수함을 조기에 탐지, 식별, 추적할 ...   
357                              K2전차의 통상명칭이다. ☞ K2 전차   

                                  출처  
0                              방사청용어  
1                              방사청용어  
2                              방사청용어  
3                              방사청용어  
4                              방사청용어  
..                               ...  
353                            방사청용어  
354  방사청용어+과학기술용어사전+국방부군수전력용어사전+KIMA  
355                            방사청용어  
356                            방사청용어  
357                            방사청용어  

[358 rows x 5 columns]

In [ ]:
import pandas as pd
weapon = pd.read_csv('/content/drive/MyDrive/Hanhwa/crawling_project/crawling_project/data/무기의세계_불필요한것 제외.csv')
weapon

용어                                                 설명
0           F-2  F-2는 일본 항공자위대가 운용중인 전투기로 F-16 전투기를 바탕으로 미국과 일본...
1    알레이버크급 구축함  알레이버크급 구축함은 미 해군의 주력 수상전투함으로 이지스함의 대명사로 알려져 있다...
2     Ju 52 수송기  Ju 52는 제2차 대전 관련 에피소드에서 약방의 감초처럼 등장하는 숨은 주인공이었...
3         X-37B  X-37B는 미 공군이 운용중인 무인우주선으로 궤도시험기라는 정식명칭을 가지고 있다...
4          미니트맨  미니트맨은 미 공군이 운용중인 대륙간탄도미사일로 미국의 핵 억제력에 중추적인 위치를...
..          ...                                                ...
361     미래 보병체계  미래 보병 체계란 미래의 보병이 휴대할 ‘디지털 군장’이다. 이런 디지털 군장을 착...
362          흑표  3.5세대 가운데서도 가장 최신 기술을 적용해 만든 것으로 평가받는 것이 한국군의 ...
363     5세대 전투기  스텔스 전투기나 폭격기를 가진 나라는 마음만 먹으면 어느 나라의 국가지휘망이라도 붕...
364    K 시리즈 소총  전쟁에서 핵심이 되는 보병에게 생명이자 가장 귀중한 친구는 무엇일까? 물론 소총이다...
365       무인전투기   사람이 직접 조종하지 않고도 공대공 또는 공대지 전투 임무를 수행하는 첨단 무인항공기.

[366 rows x 2 columns]

# 국방백서.pdf 크롤링

In [ ]:
# !pip install PyMuPDF
import fitz

In [ ]:
def extract_text_from_pdf(pdf_path: str) -> str :

    doc = fitz.open(pdf_path)    # PDF 열기
    full_text = []

    for page_num, page in enumerate(doc, start=1):
        text = page.get_text()   # 페이지의 텍스트 추출
        full_text.append(f"{text}\n")

    doc.close()
    return "".join(full_text)


if __name__ == "__main__":
    pdf_file = "/content/drive/MyDrive/Hanhwa/crawling_project/crawling_project/data/참고_국방백서 2022.pdf"  # 텍스트를 추출할 PDF 파일 경로
    extracted_text = extract_text_from_pdf(pdf_file)
    # print(extracted_text)

In [ ]:
import re
def split_sentences(text):
    # 마침표·물음표·느낌표 뒤에 공백이 올 때만 분리
    sentences = re.split(r'(?<=[\.!?])\s+', text)
    # 짧은 조각(20자 미만)은 제거
    sentences = [s for s in sentences if len(s) > 10]
    return sentences

In [ ]:
example = split_sentences(extracted_text)
for i in range(len(example)):
    example[i] = example[i].replace('\n','')

In [ ]:
print(len(example))
example[0]

3645


'발간등록번호 11-1290000-000440-11발간사오늘날 우리 군은 지난 70여 년과 다름없이 당면한 위협으로부터 국가를 보위하는 사명을 지속 수행하는 가운데, 급변하는 안보환경에 효과적으로 대처하기 위해 도약적인 혁신을 이뤄내야 하는 중대한 전환점을 맞고 있습니다.'

In [ ]:
# def clean_pdf_text(text_chunks):
#     # 1) 리스트를 하나의 문자열로
#     text = ' '.join(text_chunks)

#     # 2) 페이지 마커 제거 (--- Page 123 --- 같은 패턴)
#     text = re.sub(r'---\s*Page\s*\d+\s*---', ' ', text)

#     # 3) 발간등록번호, 긴 숫자열 제거 (필요에 따라 패턴 조정)
#     text = re.sub(r'\b\d{2,}-\d{2,}-\d{2,}\b', ' ', text)

#     # 4) 여러 줄바꿈·탭·스페이스를 하나의 스페이스로
#     text = re.sub(r'\s+', ' ', text).strip()

#     return text

In [ ]:
# cleaned = clean_pdf_text(extracted_text)
# # cleaned

In [ ]:
# example_list = split_sentences(cleaned)
# example_list[0]

'발간등록번호 11 - 1290000 - 000440 - 11 발간사오늘날우리군은지난 70 여년과다름없이당면한위협으로부터국가를보위하는사명을지속수행하는가운데 , 급변하는안보환경에효과적으로대처하기위해도약적인혁신을이뤄내야하는중대한전환점을맞고있습니다 .'

- 매핑시킬 사전 단어

In [ ]:
weapon = pd.read_csv('/content/drive/MyDrive/Hanhwa/crawling_project/crawling_project/data/무기의세계_불필요한것 제외.csv')
weapon.head()

용어                                                 설명
0         F-2  F-2는 일본 항공자위대가 운용중인 전투기로 F-16 전투기를 바탕으로 미국과 일본...
1  알레이버크급 구축함  알레이버크급 구축함은 미 해군의 주력 수상전투함으로 이지스함의 대명사로 알려져 있다...
2   Ju 52 수송기  Ju 52는 제2차 대전 관련 에피소드에서 약방의 감초처럼 등장하는 숨은 주인공이었...
3       X-37B  X-37B는 미 공군이 운용중인 무인우주선으로 궤도시험기라는 정식명칭을 가지고 있다...
4        미니트맨  미니트맨은 미 공군이 운용중인 대륙간탄도미사일로 미국의 핵 억제력에 중추적인 위치를...

In [ ]:
len(weapon['용어'].unique())

366

- 매핑단계

In [ ]:
example_list = example
len(example_list)

3645

In [ ]:
term_list = list(weapon['용어']) # example_list : 용례 리스트

In [ ]:
len(term_list)

366

In [ ]:
map_dict = {}
not_map_dict = {}
for term in term_list :
    for example in example_list :
        if term in example :
            map_dict.setdefault(term, []).append(example)
        else :
            not_map_dict[term] = 0

In [ ]:
# map_dict

In [ ]:
len(map_dict) # 366개의 단어중 39개의 단어만 용례 존재

39

In [ ]:
e = list(map_dict.keys()) # 국방백서 2022년

In [ ]:
a = list(map_dict.keys()) # 국방백서 2020년

In [ ]:
b = list(map_dict.keys()) # 국방백서 2018년

In [ ]:
c = list(map_dict.keys()) # 국방백서 2016년

In [ ]:
d = list(map_dict.keys()) # 국방백서 2016년

In [ ]:
A = set(a)
B = set(b)
C = set(c)
D = set(d)
E = set(e)

# 전체 합집합
total_union = A.union(B, C, D, E)

print(f"전체 합집합의 원소 수: {len(total_union)}")
print(total_union)

전체 합집합의 원소 수: 50
{'정찰위성', '5세대 전투기', 'F-2', '잠수함발사 탄도미사일', '대함미사일', '상륙돌격장갑차', '독도함', '현무 미사일', '공격기', '화차', '아파치', '기뢰', 'F-5 전투기', '함포', '전략 폭격기', '장갑차', '현궁', 'IED', '정찰기', 'J-20', '천궁', '박격포', '킬체인', 'K-9 자주포', '천무', '경기관총', '지뢰', '스커드미사일', '야시장비', '알레이버크급 구축함', '사드', '미사일 방어체계', '버지니아급 잠수함', '수리온', '대륙간탄도미사일', '장사정포', '연안전투함', '전자전기', 'T-50', 'P-3C', '대형 수송기', '무인전투기', '차륜형 장갑차', '강습상륙함', 'SSGN', '공중급유기', '방탄조끼', '근접전투', '울산급 호위함', '어뢰'}


In [ ]:
# 1) map_dict을 빈 리스트로 미리 초기화
map_dict = {term: [] for term in term_list}

# 2) example 한 개씩 보면서 term_list 전체를 검사
for example in example_list:
    for term in term_list:
        if term in example:
            map_dict[term].append(example)

# 3) example에 등장하지 않은 term들만 골라내기
not_map_dict = {term: 0 for term, exs in map_dict.items() if not exs}

print(len(map_dict))     # 358
print(len(not_map_dict)) # 빈 리스트인 term 개수

366
327


In [ ]:
import csv
import json

with open('/content/drive/MyDrive/Hanhwa/crawling_project/crawling_project/data/weapon_terms_usage.csv', 'w', newline='', encoding='utf-8-sig') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(['Term', 'Example Sentences'])  # 헤더

    for term, sentences in map_dict.items():
        # 리스트를 JSON 문자열로 변환해 저장 (예: ["문장1", "문장2", ...])
        sentence_list_str = json.dumps(sentences, ensure_ascii=False)
        writer.writerow([term, sentence_list_str])

In [ ]:
import csv
with open('/content/drive/MyDrive/Hanhwa/crawling_project/crawling_project/data/output_before_cleaning.csv', 'w', newline='', encoding='utf-8-sig') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(['Label', 'Text'])  # 헤더
    for label, texts in map_dict.items():
        for text in texts:
            writer.writerow([label, text])

# 텍스트 추출 후, 용례 매칭 함수

In [ ]:
import os
import re
import fitz  # PyMuPDF
import pandas as pd

# PDF 텍스트 추출 함수
def extract_text_from_pdf(pdf_path):
    doc = fitz.open(pdf_path)
    full_text = []
    for page in doc:
        text = page.get_text()
        full_text.append(text)
    doc.close()
    return "\n".join(full_text)

# 문장 단위로 나누는 함수
def split_sentences(text):
    sentences = re.split(r'(?<=[\.!?])\s+', text)
    return [s.replace('\n', '') for s in sentences if len(s) > 10] # 길이가 10 이상인 문장만 가져옴

# 여러 PDF 파일에서 텍스트 추출
def extract_all_sentences(pdf_folder):
    all_sentences = []
    for filename in os.listdir(pdf_folder):
        if filename.endswith(".pdf"):
            pdf_path = os.path.join(pdf_folder, filename)
            print(f"📄 처리 중: {filename}")  # 현재 진행 중인 PDF 출력
            text = extract_text_from_pdf(pdf_path)
            sentences = split_sentences(text)
            all_sentences.extend(sentences)
    return list(set(all_sentences))  # 중복 제거 => 다른 pdf상에 같은 문장이 있는 경우, 중복제거 => 하지만, 쉼표나 "및"과 같은 말이 추가되어 같은 문장으로 인식하지 못하지만 결국 거의 같은 문장은 제거 못함
    # ex: 1) 수상전력은 유도탄정, 어뢰정, 경비정, 화력 지원정 등 대부분 소형 고속함정으로 구성되어 있으며, 지상작전과 연계하여 지상군의 진출 지원과 연안 방어 등의 임무를 수행한다
    #     2) 수상전력은 유도탄정, 어뢰정, 소형경비정, 화력지원정 등 대부분 소형 고속함정으로 구성되어 있으며, 지상작전과 연계하여 지상군 진출을 지원하고 연안 방어 등의 임무를 수행한다.

# 용례 매핑 함수
def extract_examples_for_terms(sentences, term_list):
    map_dict = {term: [] for term in term_list}
    for sentence in sentences:
        for term in term_list:
            if term in sentence:
                map_dict[term].append(sentence)
    # 중복 문장 제거
    for term in map_dict:
        map_dict[term] = list(set(map_dict[term]))
    return map_dict

# CSV 저장 함수
def save_examples_to_csv(map_dict, output_path):
    df = pd.DataFrame([
        {"용어": term, "용례": examples}
        for term, examples in map_dict.items()
    ])
    df.to_csv(output_path, encoding="utf-8-sig", index=False)

# === 실행 ===
if __name__ == "__main__":
    # PDF 폴더 경로
    pdf_folder = "/content/drive/MyDrive/Hanhwa/crawling_project/crawling_project/data"

    # 무기 용어 리스트
    weapon = pd.read_csv("/content/drive/MyDrive/Hanhwa/crawling_project/crawling_project/data/무기의세계_불필요한것 제외.csv")
    term_list = list(weapon['용어'])

    # 전체 문장 수집
    all_sentences = extract_all_sentences(pdf_folder)

    # 용어별 용례 추출
    map_dict = extract_examples_for_terms(all_sentences, term_list)

    # CSV로 저장
    save_examples_to_csv(map_dict, "/content/drive/MyDrive/Hanhwa/crawling_project/crawling_project/data/weapon_examples.csv")

    print("✅ CSV 저장 완료: weapon_examples.csv")

📄 처리 중: 참고_국방백서 2022.pdf
📄 처리 중: 국방백서_2018.pdf
📄 처리 중: 국방백서_2014.pdf
📄 처리 중: 국방백서_2016.pdf
📄 처리 중: 국방백서_2020.pdf
📄 처리 중: 국방백서_2004.pdf
📄 처리 중: 국방백서_2008.pdf
📄 처리 중: 국방백서_2006.pdf
📄 처리 중: 국방백서_2012.pdf
✅ CSV 저장 완료: weapon_examples.csv


In [ ]:
total_examples = sum(len(exs) for exs in map_dict.values())
print(f"🔢 전체 용례 개수: {total_examples}")

# 용어별 용례 개수 저장 (선택)
count_df = pd.DataFrame([
    {"용어": term, "용례 개수": len(examples)}
    for term, examples in map_dict.items()
])
print('용례가 있는 용어 개수:' , len(count_df[count_df['용례 개수']>0]))
print('용례가 없는 용어 개수:' , len(count_df[count_df['용례 개수']==0]))
count_df[count_df['용례 개수']>0]
# count_df.to_csv("weapon_example_counts.csv", encoding="utf-8-sig", index=False)
# print("📊 용례 개수 저장 완료: weapon_example_counts.csv")

🔢 전체 용례 개수: 639
용례가 있는 용어 개수: 60
용례가 없는 용어 개수: 306


용어  용례 개수
0            F-2     29
1     알레이버크급 구축함      1
8             천무      5
27         Su-30      1
31            신궁      1
34          J-20      5
37            기뢰     51
47          SM-3      5
51          정찰위성     33
64       차륜형 장갑차      1
79       T-62 전차      1
81         강습상륙함      1
82     버지니아급 잠수함      1
86      대륙간탄도미사일     24
100          IED      3
102          공격기      9
109         SSGN      1
114           현궁      1
119  잠수함발사 탄도미사일     15
121         경기관총      2
127           사드     16
146         야시장비      2
154          정찰기     21
155          견인포      1
156       전략 폭격기      1
160       대형 수송기      1
161     미사일 방어체계     32
166           지뢰     52
187         전자전기      5
196          박격포     13
211         근접전투      4
214        공중급유기     40
216          킬체인     33
220           함포      3
228         장사정포     27
231           어뢰     17
255      F-5 전투기      1
271         A-10      2
278           화차      2
280       현무 미사일      1
291          수리온      3
302      상륙돌격장갑차      8
303          독도함      7
306      울산급 호위함      1
307           천궁     13
317         P-3C      2
319         피스아이      1
323         T-50     17
327        연안전투함      1
340       스커드미사일      1
348         방탄조끼      1
350          장갑차     69
351        대함미사일     11
353          저격수      2
357      K-9 자주포     18
359          아파치      9
360        세종대왕함      2
362           흑표      5
363      5세대 전투기      3
365        무인전투기      1

## 결과물:2004~2020년 국방백서.pdf 용례추출
- 무기의 세계 단어 : 366개
- 용례가 있는 무기의 세계 단어 : 60 개
- 전체 용례 개수 : 639개

## 고려 사항: -> pdf text 추출 상, 용례가 더럽게 가져와짐
- 용례 전처리 작업: 클렌징(불용어제거 등), 용어와 매칭되는 정확한 용례만 남기기(ex:'F-2' : 'F-22...','KF-21,,,' 로 F-2가 포함된 문장을 모두 가져옴),

- 도표에 용어가 포함되어 있는경우도 용례가 아닌 용어 나열식으로 가져옴
(ex: '연안전투함정' : 연안전투함정(척)소해함(척)상륙지휘함·상륙함(척)상륙정(척)지원함(척)고정익 항공기(대)헬기(대))
- 용례 중복 제거:
비슷한 문장이지만 완전히 같은 문장이 아니라 중복 문장으로 인식하지 못하는경우 존재

ex: 같은 문장이지만, "소형" 이라는 단어가 추가되어 중복이라 여기지 X
- 1) 수상전력은 유도탄정, 어뢰정, 경비정, 화력 지원정 등 대부분 소형 고속함정으로 구성되어 있으며, 지상작전과 연계하여 지상군의 진출 지원과 연안 방어 등의 임무를 수행한다
- 2) 수상전력은 유도탄정, 어뢰정, 소형경비정, 화력지원정 등 대부분 소형 고속함정으로 구성되어 있으며, 지상작전과 연계하여 지상군 진출을 지원하고 연안 방어 등의 임무를 수행한다.

#### 추가 전처리 필요 => 'F-2' : F-22, KF-21 등 포함되기만 하면 다 가져옴
#### 불용어 제거, 용례중복제거 등 클렌징 작업 필요

In [ ]:
map_dict

{'F-2': ['F-22, F-35 등 스텔스 전투기, P-8 포세이돈 해상초계기, 버\n지니아급 핵잠수함, 항공모함, 전략폭격기 등 해·공군 핵심전력을 인도·태평양 지역에 \n배치하고 있다.',
  '공중전력 분야는 전방위 위협에 대한 주도적 억제 및 대응이 가능하도록 은밀침투, \n장거리 정밀타격 능력을 구축하고 작전지역 내 공중우세를 달성하기 위해 차기전투기\n(F-X 2차), 한국형전투기(KF-21), F-15K 성능개량, 대형수송기, 전천후 정밀유도무기 \n등을 점진적으로 확보해 나갈 것이다.',
  '인도네시아와는 ‘특별전략적 동반자 관계’를 바탕으로 차세대 전투기(KF-21/\nIndonesian Fighter eXperimental) 공동개발, 고위급 교류, 대테러 연합훈련 등 국방·\n방산 분야에서 활발하게 협력하고 있다.',
  '공중분\n야는 방위권역 내 공중우세확보와 전략표적에 대한 타격능력 제고를 위해 KF-21 전력\n화, KF-16·F-15K 성능개량을 추진할 것이다.',
  '이에 한미동맹은 美 B-52·B-1B 장거리 폭\n격기와 F-22 스텔스 전투기, 핵잠수함, 항모전단 등 미국 전략자산의 적\n시적 전개를 통해 전략적 억제를 시행하고 있다.',
  '현재 연합지휘구조\n韓 국방부\nSCM\n美 국방부\n美 합참\n韓 합참\n연합구성군사령부\n연합군사령부 \n연합참모\nMCM\n지정된 한국군 부대\n★★★★\n미군\n사령관\n★★★\n한국군\n부사령관\n지정된 미군 부대\n저작권 전환 이후 연합지휘구조(안)\n韓 국방부\nSCM\n美 국방부\n美 합참\n韓 합참\n연합구성군사령부\n연합군사령부 \n연합참모\nMCM\n지정된 한국군 부대\n★★★★\n한국군\n사령관\n★★★\n미군\n부사령관\n지정된 미군 부대\nB-52와 F-22 한반도 전개\n\n--- Page 309 ---\n308    2022 국방백서\n한미 정상회담 (2022.5.21.)\n양 정상은 공동의 정치, 경제, 안보, 그\n리고 양국 국민 간 유대를 심화